packages/cli/src/commands/generate.ts

```diff
@@ -87,8 +87,23 @@ async function cleanDescriptorsPackage(path: string) {
     )
   }
 
-  const packageJson = await readPackage()
-  const packageSource = `file:${path}`
+  const [packageJson, packageManager] = await Promise.all([
+    readPackage(),
+    getPackageManager(),
+  ])
+
+  const protocol = await (async () => {
+    if (packageManager !== "yarn") {
+      return "file"
+    }
+
+    const yarnVersion = await getNpmVersion(packageManager)
+    const yarnMajorVersion = Number(yarnVersion.split(".").at(0))
+
+    return yarnMajorVersion >= 2 ? "portal" : "file"
+  })()
+
+  const packageSource = `${protocol}:${path}`
```

Comment:
I think it will be better if this is moved out into a separate function, so this body ends up like

```ts
  const packageJson = await readPackage()
  const protocol = await getPackageProtocol()
  const packageSource = `${protocol}:${path}`
```

(I don't mind merging both awaits with a Promise.all, it sacrifices readability but wins on performance. I'm undecided on this one 😄)

Reply:
Hey, that make sense, I've extracted this to a function. For the `Promise.all` part, for me that is perfectly readable and much more performance, but do let me know if you decide otherwise 🙏